client/src/components/advancedSearch/Dialog.tsx

```diff
@@ -0,0 +1,227 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  makeStyles,
+  Theme,
+  createStyles,
+  Typography,
+  Button,
+  IconButton,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+} from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import CachedIcon from '@material-ui/icons/Cached';
+import ConditionGroup from './ConditionGroup';
+import CopyBtn from './CopyButton';
+
+interface DialogProps {
+  others?: object;
+  open: boolean;
+  onClose: () => void;
+  onSubmit: (data: any) => void;
+  onReset: () => void;
+  onCancel: () => void;
+  title: string;
+  fields: Field[];
+  handleConditions: any;
+  conditions: any[];
+  isLegal: boolean;
+  expression: string;
+}
+
+interface Field {
+  name: string;
+  type: 'int' | 'float';
+}
+
+const AdvancedDialog = React.forwardRef((props: DialogProps, ref) => {
+  const {
+    open,
+    onClose,
+    onSubmit,
+    onReset,
+    onCancel,
+    handleConditions,
+    conditions: flatConditions,
+    isLegal,
+    expression: filterExpression,
+    title,
+    fields,
+    ...others
+  } = props;
+  const { addCondition } = handleConditions;
+  const classes = useStyles();
+
+  useEffect(() => {
+    flatConditions.length === 0 && addCondition();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // const childRef: any = useRef();
+
+  return (
+    <>
+      <Dialog
```

Comment:
we already have a similar component `DialogTemplate`, maybe we can adjust it instead of writing a new one